bindings/python/manifold3d.cpp

```diff
@@ -459,111 +486,90 @@ PYBIND11_MODULE(manifold3d, m) {
                   "marking sets of triangles that can be looked up after "
                   "further operations. Assign to MeshGL.runOriginalID vector");
 
-  py::class_<Mesh>(m, "Mesh")
+  py::class_<MeshGL>(m, "MeshGL")
       .def(
-          py::init([](py::array_t<float> &vertPos, py::array_t<int> &triVerts,
-                      std::optional<py::array_t<float>> &vertNormal,
-                      std::optional<py::array_t<float>> &halfedgeTangent) {
-            auto vertPos_view = vertPos.unchecked<2>();
-            auto triVerts_view = triVerts.unchecked<2>();
-            if (vertPos_view.shape(1) != 3)
-              throw std::runtime_error("Invalid vert_pos shape");
-            if (triVerts_view.shape(1) != 3)
-              throw std::runtime_error("Invalid tri_verts shape");
-            std::vector<glm::vec3> vertPos_vec(vertPos_view.shape(0));
-            std::vector<glm::ivec3> triVerts_vec(triVerts_view.shape(0));
-            for (int i = 0; i < vertPos_view.shape(0); i++)
-              for (const int j : {0, 1, 2})
-                vertPos_vec[i][j] = vertPos_view(i, j);
-            for (int i = 0; i < triVerts_view.shape(0); i++)
-              for (const int j : {0, 1, 2})
-                triVerts_vec[i][j] = triVerts_view(i, j);
+          // note that reshape requires mutable array_t, but this will not
+          // affect the original array passed into the function
+          py::init([](py::array_t<float> &vertProp, py::array_t<int> &triVerts,
+                      const std::optional<std::vector<uint32_t>> &mergeFromVert,
```

Comment:
Why are some of these `std::vector` and some `py::array_t`? If I understand https://pybind11.readthedocs.io/en/stable/advanced/cast/stl.html and https://pybind11.readthedocs.io/en/stable/advanced/pycpp/numpy.html correctly, the first makes a copy and the second does not?

Reply:
Yes. Btw I just noticed that I forgot to do move for the vectors.